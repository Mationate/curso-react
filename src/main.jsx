import React from 'react';
import './styles.css';
import ReactDOM, { createRoot } from 'react-dom/client';
import { FirstApp } from './FirstApp';
import { HelloWorldApp } from './HelloWorldApp';
import {CounterApp} from './CounterApp';

ReactDOM.createRoot(document.getElementById('root') ).render(
    <React.StrictMode>
        <CounterApp value={0}  />
    </React.StrictMode>
);