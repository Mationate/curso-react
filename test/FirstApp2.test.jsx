import {  render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp";

describe('pruebas en FirstApp', () => { 
    const title = 'Hola, soy mati';
    const subtitle= 'hola, soy subtitulo';
test('Debe hacer match con snapshot', () => { 
    const{container}=render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
 })

 test('should mostrar mensaje "Hola, soy mati"', () => { 
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
    // screen.debug();
  })
  test('should mostrar titulo en un h1', () => { 
    render(<FirstApp title={title} />);
    expect(screen.getByRole('heading',{level: 1}).innerHTML).toContain(title);
   })
   test('should mostrar subtitlo enviado por props', () => { 
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect(screen.getAllByText(subtitle).length).toBe(2);

    
    })

 })