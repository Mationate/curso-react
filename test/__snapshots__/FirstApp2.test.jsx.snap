// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en FirstApp Debe hacer match con snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy mati
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Mati
  </p>
</div>
`;
